components/steering_wheel/include/IO/IO_Digital.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * IO_Digital.h
+ * Digital IO Module Header
+ * @note To transfer this module to another system, the digitalInput_E and the
+ *      typedef'd IO_Digital_Flag_t must be modified to be congruent with the new system
+ */
+
+#pragma once
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "Types.h"
+#include "stm32f103xb.h" /**< Needed for GPIO_TypeDef */
+
+// System Includes
+#include <assert.h>
+
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum
+{
+    SW1 = 0U,
+    SW2,
+    SW4,
+    SW5,
+    BTN_TOP_LEFT,
+    BTN_MIDDLE,
+    BTN_TOP_RIGHT,
+    NUM_INPUTS
+} digitalInput_E;
+
+typedef uint8_t IO_Digital_Flag_t; /**< Current Implementation can handle upto 64 inputs */
+static_assert(NUM_INPUTS <= (sizeof(IO_Digital_Flag_t) * 8), 
+    "Size of DigitalInput_E must be less than or equal to the width of IO_Digital_Flag_t");
+
+typedef struct
+{
+    GPIO_TypeDef* gpiox;
+    uint16_t      pin;
+} IO_Digital_Input_S;
+
+typedef struct
+{
+    const IO_Digital_Input_S inputs[NUM_INPUTS];
```

Comment:
This shouldn't be part of the global struct, it should be internal to the module.